src/WCSimConstructRealisticPlacement.cc

```diff
@@ -0,0 +1,1168 @@
+    // - - - - pmt_offset_placement
+    // - - - - multipmt_offset_placement
+
+    // Six barrel side/bottom blocks are placed rotated around the barrel.
+    // Each PMT_only or PMT_mPMT row has frame cells (possible PMT holes) across.
+
+    // pmt_offset_placement/multipmt_offset_placement/pmtod_offset_placement
+    // ************************
+
+    // First make a single cell assembly, this will put the PMT flush against the wall
+    // Since assembly volumes just expand relative to their center point the means
+    // all nested volumes will also have the PMT againt the tyvek.
+    G4double InnerTyvekRadius = config.InnerDetectorOuterRadius;
+
+    G4ThreeVector pmt_central_position = G4ThreeVector(config.InnerDetectorOuterRadius-59.62*mm,0.0,0.0);
+    G4ThreeVector mpmt_central_position = G4ThreeVector(config.InnerDetectorOuterRadius-59.62*mm,0.0,0.0);
+    G4ThreeVector pmt_central_offset = G4ThreeVector(0.0,0.0,0.0); // -> Can be used for relative offset!
+    G4RotationMatrix* pmt_central_rotation = new G4RotationMatrix;
+    pmt_central_rotation->rotateX(90*deg);
+    pmt_central_rotation->rotateZ(270*deg);
+
+    G4RotationMatrix* multipmt_central_rotation = new G4RotationMatrix;
+    multipmt_central_rotation->rotateX(90*deg);
+    multipmt_central_rotation->rotateZ(270*deg);
+
+    // For the OD PMT we instead need to make it face outwards from the white tyvek in the OD.
+    G4ThreeVector odpmt_central_position = G4ThreeVector(config.WhiteTyvekOuterRadius+2*cm,0.0,0.0);
+    G4RotationMatrix* pmtod_central_rotation = new G4RotationMatrix;
+    pmtod_central_rotation->rotateY(270*deg);
+
+    auto pmt20_offset_placement = new G4AssemblyVolume();
+    pmt20_offset_placement->AddPlacedVolume(pmt20_dummy_logic, pmt_central_position, pmt_central_rotation);
+
+    auto pmtmulti_offset_placement = new G4AssemblyVolume();
+    pmtmulti_offset_placement->AddPlacedVolume(pmtmulti_dummy_logic, mpmt_central_position, multipmt_central_rotation);
+
+    auto pmtod_offset_placement = new G4AssemblyVolume();
+    pmtod_offset_placement->AddPlacedVolume(pmtod_dummy_logic, odpmt_central_position, pmtod_central_rotation);
+
+
+    // PMT_only_two_cell/PMT_hybrid_three_cell/PMT_od_central_cell
+    // ************************
+
+    // Now our job is to make a cell. The following diagrams show locations (#,O,M,F) = (None, 20inch, multiPMT, OD)
+    //
+    // PMT_only_two_cell
+    // _______
+    // |     |
+    // | # O |
+    // |     |
+    // | O # |
+    // _______
+
+    // PMT_hybrid_three_cell
+    // _______
+    // |     |
+    // | # O |
+    // |     |
+    // | O M |
+    // _______
+
+    // PMT_od_central_cell
+    // _______
+    // |     |
+    // | # # |
+    // |     |
+    // | #M# |
+    // _______
+
+    G4double RowSeperation = config.RowSeperation;
+
+    int NSpacesInBlock = config.NSpacesInBlock;
+    int NSegments = config.NBlocksAround * NSpacesInBlock;
+    int NBlocksUp = config.NBlocksUp;
```

Comment:
```suggestion
    //int NBlocksUp = config.NBlocksUp;
```
Unused variable compilation warning